File: ios/SagaiSambaandh/Views/ClansView.swift
import SwiftUI

struct ClansView: View {
    @State private var selectedClan: Clan? = nil
    
    private let columns = [
        GridItem(.adaptive(minimum: 160), spacing: 15)
    ]
    
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                // Header block
                VStack(alignment: .leading, spacing: 6) {
                    Text("Rajput Lineage Directory")
                        .font(BrandFonts.displayBold(size: 26))
                        .foregroundColor(.royalMaroon)
                    
                    Text("Explore the historic clans, ancestral gotras, and kingdoms of Rajasthan's royalty.")
                        .font(BrandFonts.body(size: 13))
                        .foregroundColor(.gray)
                }
                .padding(.horizontal, 20)
                .padding(.top, 15)
                
                // Grid of Clans
                LazyVGrid(columns: columns, spacing: 15) {
                    ForEach(MockData.clans) { clan in
                        ClanCard(clan: clan) {
                            selectedClan = clan
                        }
                    }
                }
                .padding(.horizontal, 20)
                
                // Heritage note block
                VStack(alignment: .leading, spacing: 10) {
                    HStack {
                        Image(systemName: "info.circle.fill")
                            .foregroundColor(.royalGold)
                        Text("About Rajput Clans")
                            .font(BrandFonts.label(size: 11))
                            .foregroundColor(.royalMaroon)
                            .fontWeight(.bold)
                    }
                    
                    Text("The Rajput community is historically divided into three major lineages (Vanshas): Suryavanshi (descended from the Solar dynasty), Chandravanshi (descended from the Lunar dynasty), and Agnivanshi (born of fire). Each clan holds native Thikanas (fortresses or estates) which shape compatibility standards.")
                        .font(BrandFonts.body(size: 12))
                        .foregroundColor(.gray)
                        .lineSpacing(4)
                }
                .padding(20)
                .background(Color.royalGold.opacity(0.05))
                .cornerRadius(12)
                .padding(.horizontal, 20)
                .padding(.top, 10)
            }
        }
        .background(Color.sandstoneIvory.edgesIgnoringSafeArea(.all))
        .navigationTitle("Rajput Clans")
        .navigationBarTitleDisplayMode(.inline)
        .sheet(item: $selectedClan) { clan in
            ClanDetailOverlay(clan: clan)
        }
    }
}

struct ClanCard: View {
    let clan: Clan
    var onTap: () -> Void
    
    private var clanBannerName: String {
        switch clan.name.lowercased() {
        case "rathore": return "caste_rajput" // Use local cached assets if possible
        case "sisodia": return "caste_maheshwari"
        case "chauhan": return "caste_brahmin"
        case "kachwaha": return "caste_oswal"
        default: return "caste_baniya"
        }
    }
    
    var body: some View {
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 0) {
                // Banner area (Simulated fort texture)
                ZStack {
                    LinearGradient(
                        colors: [.royalMaroon, Color(hex: "#2B1810")],
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    )
                    
                    Text(clan.name.prefix(1))
                        .font(BrandFonts.displayBold(size: 40))
                        .foregroundColor(.sandstoneIvory.opacity(0.2))
                }
                .frame(height: 100)
                .cornerRadius(10, corners: [.topLeft, .topRight])
                
                // Description block
                VStack(alignment: .leading, spacing: 6) {
                    Text(clan.name)
                        .font(BrandFonts.displayBold(size: 16))
                        .foregroundColor(.royalMaroon)
                    
                    Text(clan.dynasty)
                        .font(BrandFonts.label(size: 8))
                        .foregroundColor(.royalGold)
                        .fontWeight(.bold)
                    
                    Text(clan.origin)
                        .font(BrandFonts.body(size: 10))
                        .foregroundColor(.gray)
                        .lineLimit(1)
                }
                .padding(12)
            }
            .background(Color.cardBackground)
            .cornerRadius(10)
            .overlay(
                RoundedRectangle(cornerRadius: 10)
                    .stroke(Color.royalGold.opacity(0.2), lineWidth: 1)
            )
            .shadow(color: Color.black.opacity(0.04), radius: 4, x: 0, y: 2)
        }
        .buttonStyle(PlainButtonStyle())
    }
}

struct ClanDetailOverlay: View {
    let clan: Clan
    @Environment(\.presentationMode) var presentationMode
    
    var body: some View {
        VStack(spacing: 0) {
            // Header block
            ZStack(alignment: .topLeading) {
                LinearGradient(
                    colors: [.royalMaroon, .deepMaroon],
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
                .frame(height: 180)
                
                VStack(alignment: .leading, spacing: 8) {
                    Button(action: { presentationMode.wrappedValue.dismiss() }) {
                        Image(systemName: "xmark.circle.fill")
                            .font(.system(size: 24))
                            .foregroundColor(.white.opacity(0.8))
                    }
                    
                    Spacer()
                    
                    Text(clan.name)
                        .font(BrandFonts.displayBold(size: 32))
                        .foregroundColor(.sandstoneIvory)
                    
                    HStack {
                        Text(clan.dynasty)
                            .font(BrandFonts.label(size: 9))
                            .foregroundColor(.lightGold)
                            .fontWeight(.bold)
                        
                        Text("•")
                            .foregroundColor(.white)
                        
                        Text(clan.origin)
                            .font(BrandFonts.body(size: 13))
                            .foregroundColor(.sandstoneIvory.opacity(0.8))
                    }
                }
                .padding(25)
            }
            
            // Detail body block
            ScrollView {
                VStack(alignment: .leading, spacing: 20) {
                    Text("CLAN HISTORICAL PROFILE")
                        .font(BrandFonts.label(size: 11))
                        .foregroundColor(.gray)
                        .tracking(1.5)
                    
                    Text(clan.history)
                        .font(BrandFonts.body(size: 14))
                        .foregroundColor(.inkBrown)
                        .lineSpacing(6)
                    
                    Divider()
                        .background(Color.royalGold.opacity(0.3))
                    
                    // Royal Crest notice
                    HStack(spacing: 15) {
                        Image(systemName: "shield.fill")
                            .font(.system(size: 36))
                            .foregroundColor(.royalGold)
                        
                        VStack(alignment: .leading, spacing: 4) {
                            Text("Lineage verified matches inside this clan")
                                .font(BrandFonts.body(size: 11, weight: .bold))
                                .foregroundColor(.inkBrown)
                            Text("All registered brides and grooms under this lineage have undergone verification seals.")
                                .font(BrandFonts.body(size: 10))
                                .foregroundColor(.gray)
                        }
                    }
                    .padding(15)
                    .background(Color.royalGold.opacity(0.06))
                    .cornerRadius(8)
                }
                .padding(25)
            }
            Spacer()
        }
        .background(Color.sandstoneIvory.edgesIgnoringSafeArea(.all))
    }
}

// Helper for corner radius extensions
extension View {
    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
        clipShape(RoundedCorner(radius: radius, corners: corners))
    }
}

struct RoundedCorner: Shape {
    var radius: CGFloat = .infinity
    var corners: UIRectCorner = .allCorners

    func path(in rect: CGRect) -> Path {
        let path = UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius))
        return Path(path.cgPath)
    }
}

File: android/SagaiSambaandh/app/src/main/java/com/sagaisambaandh/ui/screens/ClansView.kt
package com.sagaisambaandh.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Security
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.sagaisambaandh.data.Clan
import com.sagaisambaandh.data.MockData
import com.sagaisambaandh.ui.theme.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ClansView(
    modifier: Modifier = Modifier
) {
    var selectedClan by remember { mutableStateOf<Clan?>(null) }
    val sheetState = rememberModalBottomSheetState()
    var isSheetOpen by remember { mutableStateOf(false) }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(SandstoneIvory.copy(alpha = 0.15f))
            .padding(16.dp)
    ) {
        // Header info
        Text(
            text = "Rajput Lineage Directory",
            color = RoyalMaroon,
            fontSize = 24.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Serif
        )
        Text(
            text = "Explore the historic clans, gotras, and dynasties of Rajasthan's royalty.",
            color = Color.Gray,
            fontSize = 13.sp,
            modifier = Modifier.padding(top = 4.dp, bottom = 16.dp)
        )

        // Lazy grid
        LazyVerticalGrid(
            columns = GridCells.Adaptive(150.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
            modifier = Modifier.weight(1f)
        ) {
            items(MockData.clans) { clan ->
                ClanCard(clan = clan) {
                    selectedClan = clan
                    isSheetOpen = true
                }
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // Heritage explanation banner
        Card(
            shape = RoundedCornerShape(12.dp),
            colors = CardDefaults.cardColors(containerColor = RoyalGold.copy(alpha = 0.05f)),
            modifier = Modifier.fillMaxWidth()
        ) {
            Row(
                modifier = Modifier.padding(16.dp),
                verticalAlignment = Alignment.Top
            ) {
                Icon(
                    imageVector = Icons.Default.Info,
                    contentDescription = null,
                    tint = RoyalGold,
                    modifier = Modifier.size(20.dp)
                )
                Spacer(modifier = Modifier.width(12.dp))
                Column {
                    Text(
                        text = "About Rajput Clans",
                        color = RoyalMaroon,
                        fontWeight = FontWeight.Bold,
                        fontSize = 12.sp
                    )
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = "The Rajput community is historically divided into three major lineages (Vanshas): Suryavanshi, Chandravanshi, and Agnivanshi. Each clan holds native Thikanas which shape compatibility standards.",
                        color = Color.Gray,
                        fontSize = 11.sp,
                        lineHeight = 15.sp
                    )
                }
            }
        }

        // Detail sheet overlay
        if (isSheetOpen && selectedClan != null) {
            ModalBottomSheet(
                sheetState = sheetState,
                onDismissRequest = { isSheetOpen = false },
                containerColor = SandstoneIvory
            ) {
                ClanDetailContent(clan = selectedClan!!)
            }
        }
    }
}

@Composable
fun ClanCard(
    clan: Clan,
    onTap: () -> Unit
) {
    Card(
        shape = RoundedCornerShape(10.dp),
        colors = CardDefaults.cardColors(containerColor = CardBackground),
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onTap),
        border = CardDefaults.outlinedCardBorder().copy(
            brush = Brush.linearGradient(listOf(RoyalGold.copy(alpha = 0.1f), RoyalGold.copy(alpha = 0.3f)))
        )
    ) {
        Column(modifier = Modifier.fillMaxWidth()) {
            // Top card header banner
            Box(
                contentAlignment = Alignment.Center,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(90.dp)
                    .background(
                        Brush.linearGradient(
                            colors = listOf(RoyalMaroon, InkBrown)
                        )
                    )
            ) {
                Text(
                    text = clan.name.take(1),
                    color = SandstoneIvory.copy(alpha = 0.2f),
                    fontSize = 48.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Serif
                )
            }

            // Text summary
            Column(modifier = Modifier.padding(10.dp)) {
                Text(
                    text = clan.name,
                    color = RoyalMaroon,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Serif
                )
                Text(
                    text = clan.dynasty,
                    color = RoyalGold,
                    fontWeight = FontWeight.Bold,
                    fontSize = 8.sp
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = clan.origin,
                    color = Color.Gray,
                    fontSize = 10.sp,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
            }
        }
    }
}

@Composable
fun ClanDetailContent(clan: Clan) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(24.dp)
    ) {
        Text(
            text = clan.name,
            color = RoyalMaroon,
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Serif
        )
        Row(
            modifier = Modifier.padding(top = 4.dp, bottom = 16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = clan.dynasty,
                color = RoyalGold,
                fontWeight = FontWeight.Bold,
                fontSize = 11.sp
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(text = "•", color = Color.Gray)
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = clan.origin,
                color = Color.Gray,
                fontSize = 13.sp
            )
        }

        Divider(color = RoyalGold.copy(alpha = 0.3f))
        Spacer(modifier = Modifier.height(16.dp))

        Text(
            text = "CLAN HISTORICAL PROFILE",
            fontSize = 11.sp,
            color = Color.Gray,
            fontWeight = FontWeight.Bold,
            letterSpacing = 1.5.sp
        )
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            text = clan.history,
            color = InkBrown,
            fontSize = 14.sp,
            lineHeight = 20.sp
        )

        Spacer(modifier = Modifier.height(24.dp))

        // Security seal tag
        Row(
            modifier = Modifier
                .background(RoyalGold.copy(alpha = 0.06f), shape = RoundedCornerShape(8.dp))
                .border(1.dp, RoyalGold.copy(alpha = 0.2f), RoundedCornerShape(8.dp))
                .padding(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                imageVector = Icons.Default.Security,
                contentDescription = null,
                tint = RoyalGold,
                modifier = Modifier.size(36.dp)
            )
            Spacer(modifier = Modifier.width(12.dp))
            Column {
                Text(
                    text = "Lineage verified matches inside this clan",
                    color = InkBrown,
                    fontWeight = FontWeight.Bold,
                    fontSize = 11.sp
                )
                Text(
                    text = "All registered brides and grooms under this lineage have undergone verification seals.",
                    color = Color.Gray,
                    fontSize = 10.sp,
                    lineHeight = 14.sp
                )
            }
        }
    }
}
